File: 0169-majority-element/solution.java
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length;
        int count = 0;
        int candidate = 0;

        for(int i=0; i<n; i++) {
            // 3 cases
            // if count == 0, update current elt as new candidate
            // set count = 1 , becoz no candidate is winning
            if(count == 0) {
                candidate = nums[i];
                count = 1;
            } else if(nums[i] == candidate) {
                // if curr == count , increase count, supporting the candidate
                count++;
            } else {
                // if curr != count, decrease count, cancellation of votes
                count--;
            }    
        }
        return candidate;
    }
}

File: 0169-majority-element/solution.swift
class Solution {
    func majorityElement(_ nums: [Int]) -> Int {
        var n = nums.count
        var count = 1
        var candidate = nums[0]

        for i in 1..<nums.count {
            if nums[i] == candidate {
                count += 1
            } else {
                count -= 1
            }

            if count == 0 {
                candidate = nums[i]
                count = 1
            }
        }
        return candidate
    }
}
